certified-connectors/DocuSignDemo/script.csx

```diff
@@ -2118,13 +2118,221 @@ public class Script : ScriptBase
+        if (!recipientCustomFieldObj.ContainsKey("listCustomFields"))
+        {
+          recipientCustomFieldObj["listCustomFields"] = new JArray();
+        }
+        JArray listCustomFieldsArray = (JArray) recipientCustomFieldObj["listCustomFields"];
+        listCustomFieldsArray.Add(new JObject
+        {
+          ["name"] = keyArray[1],
+          ["value"] = value,
+          ["show"] = "true"
+        });
+        continue;
+      }
+      if (string.Equals(keyArray[0], "Text Custom Fields", StringComparison.OrdinalIgnoreCase))
+      {
+        if (!newBody.ContainsKey("customFields"))
+        {
+          newBody["customFields"] = new JObject();
+        }
+        JObject recipientCustomFieldObj = (JObject) newBody["customFields"];
+        if (!recipientCustomFieldObj.ContainsKey("textCustomFields"))
+        {
+          recipientCustomFieldObj["textCustomFields"] = new JArray();
+        }
+        JArray textCustomFieldsArray = (JArray) recipientCustomFieldObj["textCustomFields"];
+        textCustomFieldsArray.Add(new JObject
+        {
+          ["name"] = keyArray[1],
+          ["value"] = value,
+          ["show"] = "true"
+        });
+        continue;
       }
+
+      // template roles parsing
+      if (!recipientMapping.ContainsKey(roleName))
+      {
+        JObject newRecipientObj = new JObject();
+        newRecipientObj["roleName"] = roleName;
+        recipientMapping[roleName] = newRecipientObj;
+      }
+      JObject recipientObj = recipientMapping[roleName];
+
+      if (keyArray.Length > 2)
+      {
+        if (!recipientObj.ContainsKey("tabs"))
+        {
+          recipientObj["tabs"] = new JObject();
+        }
+        var tabType = keyArray[1];
+        var tabLabel = keyArray[2];
+        if (string.Equals(tabType, "checkboxTabs", StringComparison.OrdinalIgnoreCase))
+        {
+          var tabObj = new JObject
+          {
+            ["tabLabel"] = tabLabel,
+            ["name"] = keyArray[3],
+            ["selected"] = value
+          };
+          JObject recipientTabs = (JObject) recipientObj["tabs"];
+          if (!recipientTabs.ContainsKey(tabType))
+          {
+            recipientTabs[tabType] = new JArray();
+          }
+          JArray recipientObjArray = (JArray) recipientTabs[tabType];
+          recipientObjArray.Add(tabObj);
+        }
+        else
+        {
+          var tabObj = new JObject
+          {
+            ["tabLabel"] = tabLabel,
+            ["value"] = value
+          };
+          JObject recipientTabs = (JObject) recipientObj["tabs"];
+          if (!recipientTabs.ContainsKey(tabType))
+          {
+            recipientTabs[tabType] = new JArray();
+          }
+          JArray recipientObjArray = (JArray) recipientTabs[tabType];
+          recipientObjArray.Add(tabObj);
+        }
+      }
+      if (string.Equals(keyArray[1], "Name", StringComparison.OrdinalIgnoreCase))
```

Comment:
Please use Case statement here

Reply:
So using switch might not be the best choice here because 1). we are not comparing a single value. We are comparing and 2). also because we are using `StringComparison.OrdinalIgnoreCase`, it would be subtly more fitting to use if case.